lib/parametric/policies.rb

```diff
@@ -1,84 +1,123 @@
+        payload.key?(key)
       end
 
-      def value
-        [@value].flatten
+      def coerce(value, key, context)
+        value
       end
 
-      protected
-      attr_reader :decorated, :options
-    end
+      def valid?(value, key, payload)
+        !payload.key?(key) || !!(value.to_s =~ @fmt)
+      end
 
-    class DefaultPolicy < Policy
-      def value
-        v = decorated.value
-        v.size > 0 ? v : Array(options[:default])
+      def meta_data
+        {}
       end
     end
+  end
 
-    class MultiplePolicy < Policy
-      OPTION_SEPARATOR = /\s*,\s*/.freeze
+  # Default validators
+  EMAIL_REGEXP = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i.freeze
 
-      def value
-        decorated.value.map do |v|
-          v.is_a?(String) ? v.split(options.fetch(:separator, OPTION_SEPARATOR)) : v
-        end.flatten
-      end
+  Parametric.policy :format, Policies::Format
+  Parametric.policy :email, Policies::Format.new(EMAIL_REGEXP, 'invalid email')
+
+  Parametric.policy :noop do
+    eligible do |value, key, payload|
+      true
     end
+  end
 
-    class NestedPolicy < Policy
-      def value
-        decorated.value.find_all{|v| v.respond_to?(:has_key?)}.map do |v|
-          options[:nested].new(v)
-        end
-      end
+  Parametric.policy :declared do
+    eligible do |value, key, payload|
+      payload.key? key
     end
+  end
 
-    class CoercePolicy < Policy
-      def value
-        decorated.value.map do |v|
-          if options[:coerce].is_a?(Symbol) && v.respond_to?(options[:coerce])
-            v.send(options[:coerce])
-          elsif options[:coerce].respond_to?(:call)
-            options[:coerce].call v
-          else
-            v
-          end
-        end
-      end
+  Parametric.policy :required do
+    message do |*|
+      "is required"
     end
 
-    class SinglePolicy < Policy
-      def value
-        decorated.value.first
-      end
+    validate do |value, key, payload|
+      payload.key? key
     end
 
-    class OptionsPolicy < Policy
-      def value
-        decorated.value.each_with_object([]){|a,arr| 
-          arr << a if options[:options].include?(a)
-        }
-      end
+    meta_data do
+      {required: true}
+    end
+  end
+
+  Parametric.policy :present do
+    message do |*|
+      "is required and value must be present"
     end
 
-    class MatchPolicy < Policy
-      def value
-        decorated.value.each_with_object([]){|a,arr| 
-          arr << a if a.to_s =~ options[:match]
-        }
+    validate do |value, key, payload|
+      case value
+      when String
+        value.strip != ''
+      when Array, Hash
+        value.any?
+      else
+        !value.nil?
       end
     end
 
+    meta_data do
+      {present: true}
+    end
+  end
+
+  Parametric.policy :gt do
+    message do |num, actual|
+      "must be greater than #{num}, but got #{actual}"
+    end
+
+    validate do |num, actual, key, payload|
+      !payload[key] || actual.to_i > num.to_i
```

Comment:
Good point. My concern would be that this policy does not coerce the value. ie. the value as passed from the outside world (ex Rails params) will most likely be a string. Not sure what the best approach is.